sda/processor_test.go

```diff
@@ -10,18 +10,102 @@ import (
 	"github.com/dedis/cothority/log"
 	"github.com/dedis/cothority/network"
 	"github.com/dedis/crypto/config"
+	"github.com/stretchr/testify/assert"
 )
 
+type basicProcessor struct {
+	msgChan chan network.Packet
+}
+
+func (bp *basicProcessor) Process(msg *network.Packet) {
+	bp.msgChan <- *msg
+}
+
+type basicMessage struct {
+	Value int
+}
+
+var basicMessageType network.MessageTypeID
+
+func TestBlockingDispatcher(t *testing.T) {
+	defer log.AfterTest(t)
+
+	dispatcher := NewBlockingDispatcher()
+	processor := &basicProcessor{make(chan network.Packet, 1)}
+
+	dispatcher.RegisterProcessor(processor, basicMessageType)
+	dispatcher.Dispatch(&network.Packet{
+		Msg:     basicMessage{10},
+		MsgType: basicMessageType})
+
+	select {
+	case m := <-processor.msgChan:
+		msg, ok := m.Msg.(basicMessage)
+		assert.True(t, ok)
+		assert.Equal(t, msg.Value, 10)
+	default:
+		t.Error("No message received")
+	}
+}
+
+func TestProcessorHost(t *testing.T) {
+	defer log.AfterTest(t)
+	h1 := newHostMock(network.Suite, "127.0.0.1:2000")
+
+	proc := &basicProcessor{make(chan network.Packet, 1)}
+	h1.RegisterProcessor(proc, basicMessageType)
+	h1.Dispatch(&network.Packet{
+		Msg:     basicMessage{10},
+		MsgType: basicMessageType})
+
+	select {
+	case m := <-proc.msgChan:
+		basic, ok := m.Msg.(basicMessage)
+		assert.True(t, ok)
+		assert.Equal(t, basic.Value, 10)
+	default:
+		t.Error("No message received on channel")
+	}
+}
+
+/*func TestAggregator(t *testing.T) {*/
+//agg := newAggregator()
+//// just aggregate individually
+//agg.aggregate(basicMessageType, &basicMessage{10})
+//agg.aggregate(basicMessageType, &basicMessage{11})
+
+//bm1 := agg.release(basicMessageType)
+//assert.Len(t, bm1, 1)
+//bm2 := agg.release(basicMessageType)
+//assert.Len(t, bm2, 1)
+//assert.Len(t, agg.release(basicMessageType), 0)
+
+//// with aggregation
+//agg.aggregate(basicMessageType, &basicMessage{12})
+//agg.aggregate(basicMessageType, &basicMessage{13})
+//agg.aggregate(basicMessageType, &basicMessage{14})
+//agg.aggregate(basicMessageType, &basicMessage{15})
+
+//agg.aggregateType(basicMessageType, 3)
+//assert.Len(t, agg.release(basicMessageType), 3)
+//assert.Nil(t, agg.release(basicMessageType), nil)
+
+//}
```

Comment:
I spend quite some time on this before letting it down. I still hoped it could be useful but then no.. :D